### DISCUSSION TITLE
OnChange Event

### DISCUSSION THREAD

**Discussion by makhdum530** [react-hook-form/react-hook-form#10906] [General] [answered]
<input
        type="text"
        id="name"
        {...register("name")}
        placeholder="Name"
      />
      <input
        type="number"
        id="phone"
        {...register("phone")}
        placeholder="Phone"
      />
      
      Phone type is number when i write  in name a text then phone value chnage from number to string

**Comment by Moshyfawn** [comment, marked as answer]
```js
register("number", { valueAsNumber: true })
```

But I advise you not to handle phone numbers as `number` but use `string` instead.

**Reply by Moshyfawn** [reply]
*native Web inputs return `string` by default
